modular_skyrat/modules/customization/modules/clothing/storage/belts.dm

```diff
@@ -0,0 +1,115 @@
+        var/datum/component/storage/STR = pouch.GetComponent(/datum/component/storage)
+        if(locked)
+            to_chat(M, "<span class='warning'>[parent] seems to be locked!</span>")
+            return FALSE
+        if(dump_destination.storage_contents_dump_act(STR, M))
+            playsound(A, "rustle", 50, TRUE, -5)
+            A.do_squish(0.8, 1.2)
+            return TRUE
+    return FALSE
+
+/obj/item/storage/belt/crusader/CtrlClick(mob/user)	//Makes ctrl-click also open the inventory, so that you can open it with full hands without dropping the sword
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.user_show_to_mob(user)
+	return
+
+/obj/item/storage/belt/crusader/AltClick(mob/user)	//This is basically the same as the normal sheath, but because there's always an item locked in the first slot it uses the second slot for swords
+	if(!user.canUseTopic(src, BE_CLOSE, NO_DEXTERITY, FALSE, TRUE))
+		return
+	if(contents.len == 2)
+		var/obj/item/drawn_item = contents[2]
+		add_fingerprint(user)
+		playsound(src, 'sound/items/unsheath.ogg', 50, TRUE, -5)
+		if(!user.put_in_hands(drawn_item))
+			to_chat(user, "<span class='notice'>You fumble for [drawn_item] and it falls on the floor.</span>")
+			update_appearance()
+			return
+		update_appearance()
+		user.visible_message("<span class='notice'>[user] takes [drawn_item] out of [src].</span>", "<span class='notice'>You take [drawn_item] out of [src].</span>")
+	else
+		to_chat(user, "<span class='warning'>[src] is empty!</span>")
+	. = ..()
+
+/obj/item/storage/belt/crusader/update_appearance(updates)
+	if(contents.len == 2)	//Checks for a sword/rod in the sheath slot, changes the sprite accordingly
+		icon_state = "crusader_belt_sheathed"
+		worn_icon_state = "crusader_belt_sheathed"
+	else
+		icon_state = "crusader_belt"
+		worn_icon_state = "crusader_belt"
+	. = ..()
+
+/obj/item/storage/belt/crusader/examine(mob/user)
+	. = ..()
+	.+= "<span class='notice'>Ctrl-click it to easily open its inventory.</span>"
+	if(contents.len == 2)	//If there's no sword/rod in the sheath slot it doesnt display the alt-click instruction
+		. += "<span class='notice'>Alt-click it to quickly draw the blade.</span>"
+		return
+
+/obj/item/storage/belt/crusader/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+
+	STR.max_items = 2
+	STR.rustle_sound = TRUE
+	STR.max_w_class = WEIGHT_CLASS_GIGANTIC	//Max size makes sure that the storage_pouch item still properly fits in it - the whitelist keeps anything fucky out
+	STR.allow_big_nesting = TRUE //Same as above
+	STR.set_holdable(list(
+		/obj/item/storage/belt/storage_pouch,
+		/obj/item/melee/sabre,
+		/obj/item/melee/cleric_mace,
+		/obj/item/nullrod	//holds any subset of nullrod in the sheath-storage, so this doesnt become a massive list
```

Comment:
Make sure there aren't any nullrod subtypes that shouldn't be fitting in here?

Reply:
![image](https://user-images.githubusercontent.com/76465278/112960343-8f4b3e80-9101-11eb-82e0-e4567d0fc6ca.png)
solved, woo

Reply:
TIL set_holdable() takes two lists, first for allowed and second for disallowed